### PR TITLE
[styles] Rename styles to default/vehicle/outdoors + light/dark

### PR DESCRIPTION
**Pull request by dvdmrtnz**
The current styles naming scheme is inconsistent:
* In the `data/styles` folder there is a `clear` style at the same level of the `vehicle` and `outdoors` styles, but then each of these styles is divided into `clear`/`night`, which is confusing as the same word `clear` is used on two different levels.

```
clear
├─ clear
├─ night
vehicle
├─ clear
├─ night
outdoors
├─ clear
├─ night
```
 
* In the code the higher level `clear` style gets no name and the lower levels use `clear`/`dark`.

```
-
├─ clear
├─ dark
vehicle
├─ clear
├─ dark
outdoors
├─ clear
├─ dark
```

* And then the debug commands use `?light`/`?dark`.

--

I propose changing the naming scheme to this:

```
default
├─ light
├─ dark
vehicle
├─ light
├─ dark
outdoors
├─ light
├─ dark
```

Light/Dark is the naming convention used by both iOS (https://developer.apple.com/design/human-interface-guidelines/dark-mode) and Android (https://developer.android.com/develop/ui/views/theming/darktheme?hl=en#change-themes) in their guidelines.

* This PR fixes section 1 of https://github.com/organicmaps/organicmaps/issues/4017

### CODE REVIEW

**Comment by dvdmrtnz** [comment]
@euf PTAL

**Comment by euf** [comment]
This is great! New naming makes a lot more sense. Thank you so much for this job, I believe it would reduce confusion for future designers a lot. 

Could you please also tag me once again after this PR is merged? I would have to update a couple of issues and wiki pages which have direct links to master branch icons.

**Comment by Jean-BaptisteC** [comment]
Can you add git blame?

**Comment by biodranik** [comment]
Thanks!

1. How does this scheme fit adding the planned variation of the more contrast dark style that is suitable for daylight use (not only for the use at night/dark environment, as the current low-contrast style)?
2. Would it be possible to make a setting "use a more contrast style" to switch it (maybe also for a light theme)?
3. Are there any benefits of using a different structure, e.g.
```
light
light/default
light/outdoor
light/vehicle
dark
dark/default
dark/outdoor
dark/vehicle
```
?
4. Would it be possible to add a new style that is light-only or dark-only?

**Comment by dvdmrtnz** [comment]
> Can you add git blame?

What do you need exactly? I don't understand

**Comment by Jean-BaptisteC** [comment]
Add your commit in .git blame ignore revs like explained here https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/

**Comment by dvdmrtnz** [comment]
> Thanks!
> 
>  1. How does this scheme fit adding the planned variation of the more contrast dark style that is suitable for daylight use (not only for the use at night/dark environment, as the current low-contrast style)?
> 
>  2. Would it be possible to make a setting "use a more contrast style" to switch it (maybe also for a light theme)?
> 
>  3. Are there any benefits of using a different structure, e.g.
> 
> 
> ```
> light
> light/default
> light/outdoor
> light/vehicle
> dark
> dark/default
> dark/outdoor
> dark/vehicle
> ```
> 
> ? 4. Would it be possible to add a new style that is light-only or dark-only?

A new `contrast` style could be added for example:

```
default
├─ light
├─ dark
vehicle
├─ light
├─ dark
outdoors
├─ light
├─ dark
contrast
├─ dark
```

If the new style for example only has dark style, a fallback in `map_style_reader.cpp` could be set to use the light style from the default style.

**Comment by dvdmrtnz** [comment]
Rebased

**Comment by pastk** [comment]
If we go a way of different color palettes for a "night" (low-contrast) theme and a "dark" (daylight suitable) theme, then we'll be able to add a "night" subfolder, e.g.
```
default
├─ light
├─ dark
├─ night
```
But this actually means that atm we have a "light" and "night" palettes and no "dark" one.

**Comment by pastk** [comment]
A different thing is the current "vehicle" style.
Its used for the car navigation mode only. And we're likely to add a "car/driving" style later.

If we'd call it "driving" indeed then its OK to leave "vehicle" as is? Or maybe better to rename to e.g. "driving-navigation" as its much more clear?

**Comment by pastk** [comment]
Re icons folders rename.

I don't think we'd ever use different icon sets for different styles (but different colors will stay for now).
So IMO it makes sense to move them outside of the current `styles/clear/style-clear/` folder.
Also its better to make separate folders for the source svg icons and many auto-generated png versions.

**Comment by dvdmrtnz** [comment]
>Also its better to make separate folders for the source svg icons and many auto-generated png versions.

They already are in different folders. Source svg icons are in `data/styles/default/light/symbols` and generated png versions are in `data/resources-mdpi_light`, `resources-hdpi_light`... folders.

**Comment by Jean-BaptisteC** [comment]
Ready to merge?

**Comment by dvdmrtnz** [comment]
Rebased and ready to merge

File: qt/CMakeLists.txt
Comment (pastk) [inline qt/CMakeLists.txt:123]: hmm why there is no outdoors style here?

File: tools/python/data/styles/setup.py
Comment (pastk) [inline tools/python/data/styles/setup.py:23]: also no outdoors style here
(what does this script do?)

File: tools/python/generate_styles_override.py
Comment (pastk) [inline tools/python/generate_styles_override.py:40]: and here

File: docs/STYLES.md
Comment (pastk) [inline docs/STYLES.md:86]: `drule_proto_default_light.bin`

File: drape/drape_tests/static_texture_tests.cpp
Comment (pastk) [inline drape/drape_tests/static_texture_tests.cpp:17]: @vng do we need the outdoors style here too?

File: drape_frontend/color_constants.cpp
Comment (pastk) [inline drape_frontend/color_constants.cpp:28]: outdoors?

**Review by pastk — APPROVED** [review]

**Comment by pastk** [comment]
> > Also its better to make separate folders for the source svg icons and many auto-generated png versions.
> 
> They already are in different folders. Source svg icons are in `data/styles/default/light/symbols` and auto-generated png versions are in `data/resources-mdpi_light`, `resources-hdpi_light`... folders.

Right, I've confused auto-generated skins with e.g. `data/styles/default/light/xhdpi` with manually created versions of bookmarks and search results icons..

Still, it'd be better to move `data/styles/default/light/symbols/` into e.g. `data/map-icons/light/svg` and `data/styles/default/light/xhdpi` into e.g. `data/map-icons/light/png/xhdpi` and `data/resources-mdpi_light` into e.g. `data/map-icons/light/skins/mdpi`. As the current structure is messy and is not logical.
We can do it in a separate PR though and discuss the best (and future-proof) structure beforehand.

File: docs/STYLES.md
Comment (dvdmrtnz) [reply docs/STYLES.md:86]: Thanks! Fixed

File: tools/python/generate_styles_override.py
Comment (dvdmrtnz) [reply tools/python/generate_styles_override.py:40]: I don't really know what these python tools do. I just changed the styles names. I don't know why there's no outdoors styles in there

File: qt/CMakeLists.txt
Comment (dvdmrtnz) [reply qt/CMakeLists.txt:123]: I would guess because the desktop application has no outdoors style?

**Comment by dvdmrtnz** [comment]
> Still, it'd be better to move `data/styles/default/light/symbols/` into e.g. `data/map-icons/light/svg` and `data/styles/default/light/xhdpi` into e.g. `data/map-icons/light/png/xhdpi` and `data/resources-mdpi_light` into e.g. `data/map-icons/light/skins/mdpi`. As the current structure is messy and is not logical. We can do it in a separate PR though and discuss the best (and future-proof) structure beforehand.

I think that's a good idea. Can be done in a separate PR

**Comment by pastk** [comment]
@vng @biodranik could you review please?

File: drape_frontend/color_constants.cpp
Comment (biodranik) [reply drape_frontend/color_constants.cpp:28]: MapStyleOutdoorsDark?

File: tools/python/generate_styles_override.py
Comment (biodranik) [reply tools/python/generate_styles_override.py:40]: @pastk it's the question to you )

**Review by biodranik — APPROVED** [review]
It is better to squash the first 4 commits into one, and remove the last one to add it after the merge.

LGTM otherwise.

File: .git-blame-ignore-revs
Comment (dvdmrtnz) [reply .git-blame-ignore-revs:5]: Will remove this commit and make a new PR after the merge

**Comment by dvdmrtnz** [comment]
Squashed and rebased

**Comment by dvdmrtnz** [comment]
> > pastk: Maybe better to merge it right after the release..

Okay. Just let me know when you are ready to merge and I can rebase the branch if necessary

File: tools/python/generate_styles_override.py
Comment (AndrewShkrob) [reply tools/python/generate_styles_override.py:40]: This script is used for designer tool. Since the designer is broken, we can ignore any inconsistency here

**Comment by pastk** [comment]
@dvdmrtnz as we're renaming e.g. `drules_proto_vehicle_clear.bin` anyway, then maybe let's make them shorter e.g. `drules_vehicle_light.bin` or put them together into a separate folder `data/drules` or maybe into respective style folders even (not sure here, probably its better to keep source and generated files separately?)
WDYT?

**Comment by pastk** [comment]
@vng please take a look when you have time.
It'd be great to finalize and merge it after the current May release is out.

File: tools/python/generate_styles_override.py
Comment (biodranik) [reply tools/python/generate_styles_override.py:40]: It would be great to revive designer back.

File: qt/CMakeLists.txt
Comment (pastk) [reply qt/CMakeLists.txt:127]: actually it has, but there is no UI for switching;
one has to type `?olight` in the search - it works well

**Comment by Jean-BaptisteC** [comment]
@rtsisyk ready to merge

**Pull request merged**
